client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -77,7 +77,7 @@
  *
  * Requests can be traced by enabling trace logging for "tracer". The trace logger outputs requests and responses in curl format.
  */
-public final class RestClient implements Closeable {
+public class RestClient implements Closeable {
```

Comment:
I know mocking http client internals require a few more lines of code, I did in some tests for es core. That said I was pretty happy cause I could leave our client final :)  @nik9000 what's your opinion on this?